extensions/neo4j/runtime/src/main/java/io/quarkus/neo4j/runtime/heath/Neo4jHealthCheck.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.neo4j.runtime.heath;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+import org.neo4j.driver.Driver;
+
+@Readiness
+@ApplicationScoped
+public class Neo4jHealthCheck implements HealthCheck {
+    @Inject
+    Driver driver;
+
+    @Override
+    public HealthCheckResponse call() {
+        HealthCheckResponseBuilder builder = HealthCheckResponse.named("Neo4j connection health check").up();
+        try {
+            driver.verifyConnectivity();
+            return builder.build();
+        } catch (Exception e) {
+            return builder.down().withData("reason", e.getMessage()).build();
```

Comment:
Not that it matters much, so just nitpicking that it is a little weird if you first call `up()` and then change it to `down()`, you could call `up()` only after `verifyConnectivity()` passes. Otherwise a big +1 from me

Reply:
This constuct is inspired by the Agroal health check because of multiple datasource support. When we will support multiple MongoDB connection/ Kafka connection /  ... this construct will be needed (because you will fails the health check if one of the multiple connection fails).
So better keep it this way.